accepted/0000-caasp-nodes.md

```diff
@@ -0,0 +1,370 @@
+At the time of writing, registering a CaaS Platform node to Uyuni/SUSE Manager has to be manually performed by the user.
+
+## Assumptions
+
+We assume that:
+* The user has already created an activation key and associated it with the onboarding. The CaaS Platform nodes are registered correctly and with the proper CaaS Platform channel(s) assigned. The
+activation key is needed to have `skuba-update` automatically patch the cluster with the latest patches available<sup>1</sup>.
+* The user is registering the CaaS Platform nodes as Salt clients
+* The registration of every CaaS Platform node to Uyuni/SUSE Manager is already completed by the user using the aforementioned methods
+
+NOTE: It is outside of this RFC to discuss automatic registration methods of the CaaS Platform nodes.
+
+## Technical description of the problem
+
+### The forbidden operations
+
+#### Patch and power management operations
+
+[`skuba-update`](https://documentation.suse.com/suse-caasp/4/single-html/caasp-admin/#_base_os_updates) systemd timer will take care of [patching](https://github.com/SUSE/skuba/blob/master/skuba-update/skuba_update/skuba_update.py#L301-L305) all installed packages locally to every node of the cluster and optionally reboot the node if `zypper` signals that a reboot is required.
+In the case that a patch that requires reboot has been installed by Uyuni/SUSE Manager before `skuba-update` applies it, then the node will not be tagged as pending a reboot.<sup>2</sup>.
+
+For the reasons above, Uyuni/SUSE Manager must not offer the possibility to issue the following operations on a registered CaaS Platform node:
+
+* Reboot a system
+* Apply a patch
+* Mark a system to automatically install patches
+* Issue any power management operation via Cobbler
+* Perform an SP migration
+
+#### Forbidden packages related operations
+
+Special handling needs to be taken with the packages that are included in the `SUSE-CaaSP-Node` pattern:
+
+- `caasp-config`
+- `cri-o`
+- `kubernetes-client`
+- `kubernetes-kubeadm`
+- `kubernetes-kubelet`
+- `patterns-base-basesystem`
+- `patterns-caasp-Node-x.y`<sup>5</sup>
+- `skuba-update`
+- `supportutils-plugin-suse-caasp`
+
+and their dependencies<sup>3</sup>. At the time of writing, it is not easy to [build the dependency tree using plain `zypper`](https://stackoverflow.com/questions/12183757/reverse-dependency-generation-with-zypper) (it will be available with `zypper` 1.14.33) and it is not clear (from the CaaS Platform side) whether a forbidden package dependency should be considered forbidden as well. For example, `iptables` is a dependency of `cri-o`, but `cri-o` does not specify which versions of `iptables` are supported.
+
+To be on the safe side, Uyuni/SUSE Manager must not offer the possibility to issue the following operations on a registered CaaS Platform node:
+
+* Upgrade a package
+* Remove a package
```

Comment:
This is fine as long as:

  * No interactive package is installed/upgraded
  * The dependencies of `patterns-caasp-Node-x.y` are not broken
  * The pacakge `patterns-caasp-Node-x.y` is not touched, the upgrade of this package should be done only via skuba